### ISSUE TITLE
Variables and bindings to Flashlight Variable functions

### ISSUE THREAD

**Issue by asilvas** [facebookresearch/shumai#83]
The only support for variables I'm seeing is the `var` function that returns a Tensor as a variable. But I'm not seeing any established patterns for generating bindings to Flashlight's variable functions (such as softmax or the many activation functions). I noticed you had merged [custom ops](https://github.com/facebookresearch/shumai/blob/main/shumai/tensor/tensor_ops.ts) instead of relying on bindings to the underlying Flashlight function (which would reduce FFI calls) and wondered what the logic was. Still navigating the intended patterns for this project.

**Comment by cryptodeal** [comment]
I'd actually asked @bwasti that same question when starting to work a bit more in the native code; the decision was made to only generate bindings for functions vs variable functions so that we are better able leverage JS runtime for stack trace, inspection, etc. 

Ultimately, I suspect this also will make it a bit easier to build and bring your own backend that can be used to generate custom FFI bindings used by Shumai. 

I could be missing additional context for this design choice, but that's my understanding. Likely @bwasti or @jacobkahn will be able to elaborate on this a bit further.

**Comment by asilvas** [comment]
Thanks for clarification. I'll close this issue with some (naive) counter points:

* Despite the low FFI overhead of Bun, the more complex functions can add up. Perhaps it's negligible, haven't tested any real-world deltas
* Might help with velocity rounding out feature parity with FL if Shumai was almost entirely a JS binding library and left all the heavy lifting to FL
* The larger and more complex that FL gets over time may lead to gaps (Tensorflow vs Tensorflow.js comes to mind, yuck!) and inconsistencies

In any case I see a bright future for Shumai (and FL), and appreciate all the efforts!

**Comment by cryptodeal** [comment]
Feel free to re-open for the sake of further discussion on this!
